src/internal/channel.js

```diff
@@ -124,16 +124,27 @@ export function eventChannel(subscribe, buffer = buffers.none(), matcher) {
   }
 
   const chan = channel(buffer)
+  const close = () => {
+    if(!chan.__closed__) {
+      if (unsubscribe) {
+        unsubscribe()
+      }
+      chan.close()
```

Comment:
yeah, that makes sense to me. i think generally the argument for doing this would be you want to unsubscribe and allow the developer to clean up everything before officially closing the channel.